analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -217,10 +217,17 @@ variableDeclarator.Parent.Parent switch
             Diagnostic CreateS1144Diagnostic(SyntaxNode syntaxNode, ISymbol symbol) =>
                 Diagnostic.Create(RuleS1144, GetIdentifierLocation(syntaxNode), accessibility, symbol.GetClassification(), GetMemberName(symbol));
 
-            static Location GetIdentifierLocation(SyntaxNode syntaxNode) =>
-                syntaxNode.GetIdentifier() is { } identifier
+            static Location GetIdentifierLocation(SyntaxNode syntaxNode)
+            {
+                if (syntaxNode is IndexerDeclarationSyntax indexer)
+                {
+                    return indexer.ThisKeyword.GetLocation();
+                }
```

Comment:
There are already cases where we do not strictly return identifier token from `GetIdentifier` (e.g. PredefinedTypeSyntax or OperatorDeclarationSyntax). I would consider the `this` in an indexer to be no different from an operator in an operator declaration and add the handling to GetIdentifier.

Reply:
Agreed, it's much cleaner. I moved it to `GetIdentifier()`